code/src/components/MessageBoard.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react'
+import moment from 'moment'
+
+import { API_URL, HEARTS_URL } from '../reusable/urls'
+
+import Loading from '../components/Loading'
+import ErrorBoard from '../components/ErrorBoard'
+import MyLikes from './MyLikes'
+
+const MessageBoard = ({ thoughts, setThoughts, myLikes, setMyLikes }) => {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+
+  const updateBoard = (url) => {
+    setLoading(true)
+    fetch(url)
+      .then(response => response.json())
+      .then(receivedThoughts => {
+        setThoughts(receivedThoughts)
+        setTimeout(() => {
+          setLoading(false)
+        }, 2000)
+      })
+      .catch(err => {
+        setLoading(false)
+        setError(true)
+      })
+  }
+
+  useEffect(() => {
+    const fetchThoughts = (url) => {
+      setLoading(true)
+      fetch(url)
+        .then(response => response.json())
+        .then(receivedThoughts => {
+          setThoughts(receivedThoughts)
+          setTimeout(() => {
+            setLoading(false)
+          }, 2000)
+        })
+        .catch(err => {
+          setLoading(false)
+          setError(true)
+        })
+    }
+    fetchThoughts(API_URL)
+  },[setThoughts]);
+
+  const onThoughtLiked = (thoughtLiked) => {
+    const updatedThoughts = thoughts.map((thought) => {
+      if (thought._id === thoughtLiked) {
+        thought.hearts += 1
+      } 
+      return thought
+    })
+    setThoughts(updatedThoughts)
+    if (myLikes.indexOf(thoughtLiked) === -1) {
+      setMyLikes([thoughtLiked, ...myLikes])
+    }
+  }
+
+  const onHeartClicked = (id) => {
+    const options = {
+      method: "POST",
+      headers: {
+        'Content-Type': 'application/json'
+      },
+    }
+    fetch(HEARTS_URL(id), options)
+      .then(res => res.json())
+      .then(() => {
+        onThoughtLiked(id)
+      })
+  }
+
+  return loading ? <Loading loading={loading} /> : error ? <ErrorBoard /> : (
+    <div className="message-board">
+      <MyLikes 
+      fetchThoughts= {updateBoard}
+      numberOfMyLikes = {myLikes.length}
+      API_URL = {API_URL}
+      />
+      {thoughts.map(thought => (
+        <div className="message-container" key={thought._id}>
+          <div className="thought-date-wrapper">
+            <p className="thought">{thought.message}</p>
+            <p className={`date ${moment(thought.createdAt).fromNow()=== "a few seconds ago"? `red`: null}`}>{moment(thought.createdAt).fromNow()=== "a few seconds ago"? "New message": moment(thought.createdAt).fromNow()} </p>
```

Comment:
I had a hard time reading this, mostly because I'm tired...but maybe it's not possible to put it on several lines, or is it?🤔  Again, ternary operators😻